kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/JvmDependenciesIndexImpl.kt

```diff
@@ -36,38 +34,8 @@ class JvmDependenciesIndexImpl(_roots: List<JavaRoot>) : JvmDependenciesIndex {
     private val maxIndex: Int
         get() = roots.size
 
-    // each "Cache" object corresponds to a package
-    private class Cache {
-        private val innerPackageCaches = HashMap<String, Cache>()
-
-        operator fun get(name: String) = innerPackageCaches.getOrPut(name, ::Cache)
-
-        // indices of roots that are known to contain this package
-        // if this list contains [1, 3, 5] then roots with indices 1, 3 and 5 are known to contain this package, 2 and 4 are known not to (no information about roots 6 or higher)
-        // if this list contains maxIndex that means that all roots containing this package are known
-        val rootIndices = IntArrayList(2)
-    }
-
-    // root "Cache" object corresponds to DefaultPackage which exists in every root. Roots with non-default fqname are also listed here but
-    // they will be ignored on requests with invalid fqname prefix.
-    private val rootCache: Cache by lazy {
-        Cache().apply {
-            roots.indices.forEach(rootIndices::add)
-            rootIndices.add(maxIndex)
-            rootIndices.trimToSize()
-        }
-    }
-
-    // holds the request and the result last time we searched for class
-    // helps improve several scenarios, LazyJavaResolverContext.findClassInJava being the most important
-    private var lastClassSearch: Pair<FindClassRequest, SearchResult>? = null
```

Comment:
Cache **№2**. There are cache hits, actually more than I thought, so this could be useful. Could be made thread safe without much problem.